### PR TITLE
Support for multiple signals

### PR DESCRIPTION
**Pull request by SeanLatimer**
Implements the feature described in #42 

Backwards compatible with previous releases

Also had to update the shutdown tests as they never errored which meant they never called the assertion.

### CODE REVIEW

File: lib/terminus.js [hunk 1]
```diff
@@ -57,33 +57,37 @@ function decorateWithHealthCheck (server, options) {
...
+          process.kill(process.pid, signal)
+        })
+        .catch((error) => {
+          logger('error happened during shutdown', error)
+          process.exit(1)
       })
+    }
   }
-  process.on(signal, cleanup)
+  signals.forEach(
```
Comment (gergelyke) [inline lib/terminus.js]: should we merge `signal` into `signals`, so the listener is only appended once?

File: lib/terminus.spec.js [hunk 2]
```diff
@@ -96,47 +96,33 @@ describe('Terminus', () => {
...
-    try {
-      execFileSync('node', ['lib/standalone-tests/terminus.onshutdown.sigterm.js'])
-    } catch (ex) {
-      expect(ex.stdout.toString().trim()).to.eql('on-sigterm-runs\non-shutdown-runs')
-      return
-    }
-
-    throw new Error('running the test should throw, as the exitcode is not 0')
+  it('runs onShutdown after SIGTERM onSignal', () => {
+    let result = spawnSync('node', ['lib/standalone-tests/terminus.onshutdown.sigterm.js'])
```
Comment (gergelyke) [inline lib/terminus.spec.js]: can we switch to const to be consistent with the rest of the tests?

File: lib/terminus.js (hunk 1 above)
Comment (SeanLatimer) [reply lib/terminus.js]: `signal` gets merged into the `signals` array on L107, just before the call to `decorateWithSignalHandler`

File: lib/terminus.spec.js (hunk 2 above)
Comment (SeanLatimer) [reply lib/terminus.spec.js]: I'll get them switched over

**Comment by SeanLatimer** [comment]
I'll get those done and push the changes shortly 👍

File: lib/terminus.js (hunk 1 above)
Comment (gergelyke) [reply lib/terminus.js]: oh cool, thanks - I missed that

File: lib/standalone-tests/terminus.onmultiple.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  signal: SIGNAL,
+  onSignal: () => {
+    console.log('on-' + SIGNAL.toLowerCase() + '-runs')
+    return Promise.resolve()
+  }
+})
+
+server.listen(8000, () => {
+  process.kill(process.pid, SIGNAL)
+})
```
Comment (gergelyke) [inline lib/standalone-tests/terminus.onmultiple.js]: can you add a new line here?

File: lib/standalone-tests/terminus.onshutdown.multiple.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    return Promise.resolve()
+  },
+  onShutdown: () => {
+    console.log('on-shutdown-runs')
+  }
+})
+
+server.listen(8000, () => {
+  process.kill(process.pid, SIGNAL)
+})
```
Comment (gergelyke) [inline lib/standalone-tests/terminus.onshutdown.multiple.js]: can you add a new line here?

**Comment by gergelyke** [comment]
thanks for the tests @SeanLatimer - just two minor things I 've commented on before merging

**Comment by SeanLatimer** [comment]
@gergelyke Ok, done. Usually I use Prettier for formatting but it was causing too much of a change for this PR.

**Comment by SeanLatimer** [comment]
Actually, it seems that after pushing the new lines are not there?

 Not sure what happened, I still have them in my editor on my end

**Comment by SeanLatimer** [comment]
There we go, they got removed on save for some reason.

**Comment by gergelyke** [comment]
thanks Sean - merged 🎉

**Pull request merged**